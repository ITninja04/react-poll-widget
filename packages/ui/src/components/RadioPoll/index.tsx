import React from 'react';
import { Form, Input } from 'semantic-ui-react';

import Checkbox from '../Checkbox';
import './styles.scss';

type RadioPollOption = {
  label: string;
  value: string;
};

const RadioPoll: React.FC<{
  options: RadioPollOption[],
  selected?: RadioPollOption,
  onSelect: (option: RadioPollOption) => void,
  onSelectCustom: () =>  void,
  onChange: (e: React.ChangeEvent) => void,
  hasCustomOption: boolean,
  customSelected: boolean
}> = ({
  options,
  selected,
  onSelect,
  onSelectCustom,
  onChange,
  hasCustomOption,
  customSelected
}) => (
      <Form>
        {
          options.map(option => (
            <Form.Field>
              <Checkbox
                type='radio'
                label={option.label}
                checked={selected !== undefined && selected.value === option.value}
                onClick={() => onSelect(option)}
              />
            </Form.Field>
          ))
        }
        {
          hasCustomOption &&
          <Form.Field>
          <Checkbox
            type='radio'
            label={
              <Input size='small' onChange={onChange} />
            }
            checked={customSelected}
            onClick={onSelectCustom}
          />
          </Form.Field>
        }
      </Form>
    )

export default RadioPoll;