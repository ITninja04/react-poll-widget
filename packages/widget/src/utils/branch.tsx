export type BranchFunction = (condition: boolean, ComponentIfTrue: React.ReactNode, ComponentIfFalse: React.ReactNode) => React.ReactNode;

export const branch: BranchFunction = (condition, ComponentIfTrue, ComponentIfFalse) => condition
  ? ComponentIfTrue
  : ComponentIfFalse;