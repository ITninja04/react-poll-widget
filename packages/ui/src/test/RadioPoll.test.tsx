import RadioPoll from "../components/RadioPoll";
import { createSnapshotTest } from "./utils";

createSnapshotTest(RadioPoll, {
  options: [
    { id: '1', label: 'Test 1', value: '1' },
    { id: '2', label: 'Test 2', value: '2' },
    { id: '3', label: 'Test 3', value: '3' }
  ]
}, 'Basic')