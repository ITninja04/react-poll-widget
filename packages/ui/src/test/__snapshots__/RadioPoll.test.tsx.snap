// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Snapshot) Basic should render 1`] = `
<DocumentFragment>
  <form
    class="ui form"
  >
    <div
      class="field"
    >
      <div
        class="poll-widget-checkbox radio"
      >
        Test 1
      </div>
    </div>
    <div
      class="field"
    >
      <div
        class="poll-widget-checkbox radio"
      >
        Test 2
      </div>
    </div>
    <div
      class="field"
    >
      <div
        class="poll-widget-checkbox radio"
      >
        Test 3
      </div>
    </div>
  </form>
</DocumentFragment>
`;
